src/lib/src/component/navigation/navigation.component.ts

```diff
@@ -71,32 +74,49 @@ export class NavigationComponent implements OnInit {
       this.retrieveWorkspaceRoot();
       this.changeDetectorRef.detectChanges();
     });
-    this.messagingService.subscribe(events.NAVIGATION_REFRESH, element => {
-      this.retrieveWorkspaceRoot();
+    this.messagingService.subscribe(events.NAVIGATION_CREATED, payload => {
+      this.handleNavigationCreated(payload);
     });
     this.messagingService.subscribe(events.NAVIGATION_SELECT, element => {
       this.uiState.selectedElement = element as WorkspaceElement;
       this.changeDetectorRef.detectChanges();
     });
   }
 
+  handleNavigationCreated(payload: any): void {
+    this.retrieveWorkspaceRoot().then(() => {
+      this.revealElement(payload.path);
```

Comment:
can payload be typed somehow?

Reply:
what if retrieval returns undefined?

Reply:
We can't type it as it differs from event to event. In `MessagingService` we defined:
```
   subscribe(type: string, callback: (payload: any) => void): Subscription;
```

Reply:
Added a cast + test for the undefined case